WalletWasabi/Transactions/TransactionBuilding/SmartCoinSelector.cs

```diff
@@ -29,23 +30,54 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
 				throw new InsufficientBalanceException(targetMoney, available);
 			}
 
-			// Group coins by scriptPubKey in order to treat them all as one.
-			var coinsByScriptPubKey = UnspentCoins
+			var uniqueClusters = UnspentCoins
+				.Select(coin => coin.Clusters)
+				.Distinct();
+
+			var clusters = (uniqueClusters.Count() < 10)
+				? uniqueClusters
+					.CombinationsWithoutRepetition(ofLength: 1, upToLength: 6)
+					.Select(clusterCombination => UnspentCoins.Where(coin => clusterCombination.Contains(coin.Clusters)))
+					.ToList()
+				: new List<IEnumerable<SmartCoin>>();
+
+			clusters.Add(UnspentCoins);
```

Comment:
I guess it doesn't matter from the final result of the function, but I think this is redundant when get the the `clusters` based on the combinations, since it'd result in the exact same combination, no?

Reply:
If I'm right, I'd do

```
var clusters;
if(uniqueClusters.Count() < 10)
{
   clusters = combinations;
}
else
{
   clusters = new list<<>>(unspentcoins)
}
```

Reply:
It matters. Given we cannot create all the possible combinations, in many cases none of the generated combinations would have enough money required by the `targetMoney` so, in those cases we have to choose the coins from the list of all the selected coins. The cleanest way to achieve that is by adding the `unspentcoins` list as the last one of all the possible combinations.

Reply:
I see, it matters. But can you address my issue, too? Aren't we duplicating the "everything in it" combination?